### PR TITLE
Mixed Authority Mode for Document Generation and small additional features

### PR DESCRIPTION
**Pull request by micha91**
In this PR we introduce the new mixed authority mode which allows the user to specify sections in a document in Polarion, which should be overwritten by the rendered content of a jinja template. The start of this section is identified by this Polarion Wiki Macro - the important parts are the div class and its ID:
```html
<div class="c2pAreaStart" id="TestArea">
#if ($document.getStatus().id == "draft")
<span style="font-weight: bold;background-color: #FFFF00;">↓↓↓Below this point all content is autogenerated and will be overwritten ↓↓↓</span>
#end
</div>
```
The end of an autogenerated section is identified by this Macro:
```html
<div class="c2pAreaEnd" id="TestArea">
#if ($document.getStatus().id == "draft")
<span style="font-weight: bold;background-color: #FFFF00;">↑↑↑Above this point all content is autogenerated and will be overwritten ↑↑↑</span>
#end
</div>
```
In addition the following minor features were implemented:
**Documents**
- We now support heading numbering using our config
- You can configure default rendering layouts for inserted workitems
  - By default we keep existing configuration when documents are updated. You can overwrite this behaviour using flags to use the settings from the config instead of from the document
- If we try to insert a workitem into a document and it already exists in the document a wanring will be logged and we insert a link instead

**WorkItems**
It's now possible to insert diagrams in a jinja template. E.g.
```
{{ insert_diagram(work_item, object.tree_view, "tree_view") }}
```
will insert the tree_view diagram of `object`. The diagram will be stored with "tree_view" as file name in the work items attachments. If there already is a diagram named tree_view this one will be used instead of generating another one. So if you use the `add_tree_diagram` serializer, the diagram will only be generated once.

### CODE REVIEW

File: tests/test_documents.py [hunk 1]
```diff
@@ -118,7 +127,11 @@ def test_update_document(
...
     assert len(new_doc.rendering_layouts) == 1
-    assert new_doc.rendering_layouts[0].properties == [{"key": "value"}]
+    assert new_doc.rendering_layouts[
+        0
+    ].properties == polarion_api.data_models.RenderingProperties(
+        fields_at_start=["ID"]
+    )
     assert (
         etree.tostring(content[0])
         .decode("utf-8")
```
Comment (ewuerger) [inline tests/test_documents.py:137]: Can you fix the format here please? Just for me :>

File: tests/test_documents.py [hunk 2]
```diff
@@ -135,3 +148,170 @@ def test_update_document(
...
+        .decode("utf-8")
+        .startswith(
+            '<h3 id="polarion_wiki macro name=module-workitem;'
+            'params=id=ATSY-18305"'
+        )
+    )
+    assert content[11].text == "Overwritten: Overwrite global param"
+    assert content[12].text == "Local Test section 2"
+    assert content[14].text == "Some postfix stuff"
+
```
Comment (ewuerger) [inline tests/test_documents.py:212]:
```suggestion
```

File: tests/test_documents.py [hunk 3]
```diff
@@ -135,3 +148,170 @@ def test_update_document(
...
+    )
+
+    content: list[etree._Element] = html.fromstring(
+        new_doc.home_page_content.value
+    )
+
+    assert len(content) == 15
+    assert content[0].tag == "h1"
+    assert content[1].tag == "p"
+    assert content[2].tag == "p"
```
Comment (ewuerger) [inline tests/test_documents.py:192]:
```suggestion
    assert [c.tag for c in content[:3]] == ["h1", "p", "p"]
```

File: tests/test_documents.py [hunk 4]
```diff
@@ -135,3 +148,170 @@ def test_update_document(
...
+    assert (
+        etree.tostring(content[4])
+        .decode("utf-8")
+        .startswith("<h3>New Heading</h3>")
+    )
```
Comment (ewuerger) [inline tests/test_documents.py:197]:
```suggestion
    assert (c4 := content[4]).tag == "h3" and c4.text == "New Heading" 
```

File: tests/test_documents.py [hunk 5]
```diff
@@ -135,3 +148,170 @@ def test_update_document(
...
+    assert content[6].text == "Local Test section 1"
+    assert content[8].text == "This will be kept."
+    assert (
+        etree.tostring(content[10])
+        .decode("utf-8")
+        .startswith(
+            '<h3 id="polarion_wiki macro name=module-workitem;'
+            'params=id=ATSY-18305"'
+        )
+    )
```
Comment (ewuerger) [inline tests/test_documents.py:208]:
```suggestion
    assert content[10].id == (
        "polarion_wiki macro name=module-workitem;'
        'params=id=ATSY-18305"'
    )
```

File: tests/test_documents.py [hunk 6]
```diff
@@ -135,3 +148,170 @@ def test_update_document(
...
+    assert conf.mixed_authority[1].instances[0].section_params == {
+        "section1": {"param_1": "Test"}
+    }
+
+
+def test_combined_config():
+    with open(COMBINED_CONFIG, "r", encoding="utf-8") as f:
+        conf = document_config.read_config_file(f)
+    assert len(conf.full_authority) == 2
+    assert len(conf.mixed_authority) == 2
```
Comment (ewuerger) [inline tests/test_documents.py:273]:
```suggestion
    with open(COMBINED_CONFIG, "r", encoding="utf-8") as f:
        conf = document_config.read_config_file(f)

    assert len(conf.full_authority) == 2
    assert len(conf.mixed_authority) == 2
```

**Review by ewuerger — CHANGES_REQUESTED** [review]
After that LGTM.

File: capella2polarion/converters/polarion_html_helper.py [hunk 7]
```diff
@@ -102,3 +108,41 @@ def check_model_element(
...
     def setup_env(self, env: jinja2.Environment):
         """Implement this method to adjust a newly created environment."""
+
+
+def remove_table_ids(
+    html_content: str | list[etree._Element],
+) -> list[etree._Element]:
+    """Remove the ID field from all tables.
+
+    This is necessary due to a bug in Polarion.
```
[the remark this answers is not in the document]
Comment (micha91) [reply capella2polarion/converters/polarion_html_helper.py:118]: Table IDs are not uniwue in Polarion, if you add them via the UI to a live-doc

**Review by ewuerger — APPROVED** [review]
LGTM

**Pull request merged**